x-pack/plugins/code/public/components/query_bar/components/query_bar.tsx

```diff
@@ -475,16 +478,16 @@ export class CodeQueryBar extends Component<Props, State> {
 }
 
 const mapStateToProps = (state: RootState) => ({
-  repoSearchResults: state.search.repositorySearchResults
-    ? state.search.repositorySearchResults.repositories
+  repoSearchResults: state.search.scopeSearchResults[state.search.scopeSearchQuery]
+    ? state.search.scopeSearchResults[state.search.scopeSearchQuery].repositories
```

Comment:
why do we want to have arrays to cache repository results for each query? is this really necessary?

Reply:
let's just add another field in the state like `repoScopeSearchResults`